### PR TITLE
docs: Improve 'template' docs, add new doc for search fields

### PR DESCRIPTION
**Pull request by NickolausDS**
Previously, the 'templates' doc was large and included a bunch of
information, making it hard to follow. This has now been split into two docs for both simple
templating and how it differs from Django, and separately the 'fields' features
for processing Globus Search data.

### CODE REVIEW

**Review by tskluzac — APPROVED** [review]
This is much clearer, thank you! 

I appreciate the explanation on including vs excluding 'myportal' in the template path, as that was confusing to me before.

**Review by sturoscy-personal — APPROVED** [review]
Much clearer. Thanks again!

**Comment by NickolausDS** [comment]
Thanks all!
